readthedocs_ext/readthedocs.py

```diff
@@ -126,6 +158,36 @@ def rtd_render(self, template, render_context):
                                                         app.builder.templates)
 
 
+def generate_json_artifacts(app, pagename, templatename, context, doctree):
+    """
+    Generate JSON artifacts for each page.
+
+    This way we can skip generating this in other build step.
+    """
+    try:
+        # We need to get the output directory where the docs are built
+        # _build/json.
+        build_json = os.path.abspath(
+            os.path.join(app.outdir, '..', 'json')
+        )
+        outjson = os.path.join(build_json, pagename + '.fjson')
+        outdir = os.path.dirname(outjson)
+        if not os.path.exists(outdir):
+            os.makedirs(outdir)
+        with open(outjson, 'w+') as json_file:
+            to_context = {
+                key: context[key]
+                for key in KEYS
+                if key in context
+            }
+            json_file.write(json.dumps(to_context, indent=4))
+            log.info('{page} processed.'.format(page=outjson))
+    except Exception:
+        log.exception(
```

Comment:
This doesn't make the build to fail (we need to throw the exception), it should fail?